testing/test_transforms.py

```diff
@@ -0,0 +1,248 @@
+from collections import namedtuple
+
+import pytest
+import numpy as np
+
+from idelib import importer
+from idelib.dataset import Dataset
+from idelib.transforms import Transform, AccelTransform, Univariate, Bivariate
+from .file_streams import makeStreamLike
+
+
+@pytest.fixture()
+def genericTransform():
+    yield Transform()
+
+
+@pytest.fixture()
+def badTransform():
+    tf = Transform()
+    tf._function = None
+    yield tf
+
+
+@pytest.fixture(
+        params=[
+            ((0,), 0),
+            ((1,), 0),
+            ((3, 1), 0),
+            ((3, 1), 5),
+            ],
+        ids=[
+            'zero-trivial',
+            'constant',
+            'simple-linear',
+            'referenced-linear',
+            ],
+        )
+def univariate(request):
+    coeffs, reference = request.param
+    tf = Univariate(coeffs, reference=reference)
+    return tf
+
+
+@pytest.fixture()
+def ssx66115():
+    testDataset = importer.openFile(makeStreamLike('./testing/SSX66115.IDE'))
+    importer.readData(testDataset)
+    yield testDataset
+    testDataset.close()
```

Comment:
The function `makeStreamLike` caches file data in memory to reduce test execution time.  As for not using `importFile`, that's mostly because `makeStreamLike` returns a stream-like object, but `importFile` call's open on the file name passed in, and it was easier to just circumvent that call.

Reply:
> The function makeStreamLike caches file data in memory to reduce test execution time.

Ah, yep that seems good

> As for not using importFile, that's mostly because ... `importFile` calls open on the file name passed in

...ew. yeah it'd be nice to change that. maybe during development on the "universal file opener" function?

Reply:
...actually now that I look at it, you can set a pytest fixture's yield/return value to be shared across multiple tests (as opposed to being called and re-generated for every individual test) [by setting the fixture's scope parameter](https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session). 

not critical, but it looks like a more idiomatic way to cache results for test fixtures. one of these days we could maybe get around to using that 🤷

Reply:
Yeah, I'll take a look, I think there's a better way to cache.  That being said, I'd point out a few issues I've found with fixtures, some of which are very specific to this case:
- The value that gets re-used is a reference, so it needs to be recalculated every time, rather than just re-passed.
- When a test uses a fixture, it also needs to be able to see every other fixture referenced by that fixture and so on.  If you have too many fixtures, then you end up with enormous blocks of imports, or `from x import *` which I don't like.  
- Adding narrow helper functions to the scope of what's "correct" in Pytest allows you to make functions to parameterize fixtures or tests, since fixtures can't be called in pytest marks.

Reply:
yeah I dug into it a bit; it's too complicated with `unittest.TestCase` classes. all good 👍